server/pulp/server/controllers/repository.py

```diff
@@ -816,6 +816,149 @@ def sync(repo_id, sync_config_override=None, scheduled_call_id=None):
+
+    :param conduit: allows the plugin to interact with core pulp
+    :type  conduit: pulp.plugins.conduits.repo_publish.RepoPublishConduit
+    :param repo_id: identifies the repo to sync
+    :type  repo_id: str
+
+    :return:  Whether a content unit has been removed since the last_sync timestamp
+    :rtype:   boolean
+    """
+    last_sync = conduit.last_sync()
+
+    if last_sync is None:
+        return False
+
+    # convert the iso8601 datetime string to a python datetime object
+    last_sync = dateutils.parse_iso8601_datetime(last_sync)
+    repo_obj = model.Repository.objects.get_repo_or_missing_resource(repo_id=repo_id)
+    last_removed = repo_obj.last_unit_removed
+
+    # check if a unit has been removed since the past sync
+    if last_removed is not None:
+        if last_removed > last_sync:
+            return True
+
+    return False
+
+
+def check_config_updated_since_last_sync(conduit, repo_id):
+    """
+    Checks whether the config has been changed since the last sync occurred.
+
+    :param conduit: allows the plugin to interact with core pulp
+    :type  conduit: pulp.plugins.conduits.repo_publish.RepoPublishConduit
+    :param repo_id: identifies the repo to sync
+    :type  repo_id: str
+
+    :return:  Whether the config has been changed since the last sync occurred.
+    :rtype:   boolean
+    """
+    last_sync = conduit.last_sync()
+
+    if last_sync is None:
+        return False
+
+    # convert the iso8601 datetime string to a python datetime object
+    last_sync = dateutils.parse_iso8601_datetime(last_sync)
+    repo_importer = model.Importer.objects.get_or_404(repo_id=repo_id)
+    # the timestamp of the last configuration change
+    last_updated = repo_importer.last_updated
+
+    # check if a configuration change occurred after the most recent sync
+    if last_updated is not None:
+        if last_sync < last_updated:
+            return True
+
+    return False
+
+
+def check_override_config_change(repo_id, call_config):
+    """
+    Checks whether the override config is different from the override config on
+    the previous sync.
+
+    :param repo_id:     identifies the repo to sync
+    :type  repo_id:     str
+    :param call_config: Plugin Call Configuration
+    :type  call_config: pulp.plugins.config.PluginCallConfiguration
+
+    :return:  Whether the override config is different from the override config on
+    the previous sync.
+    :rtype:   boolean
+    """
+    repo_importer = model.Importer.objects.get_or_404(repo_id=repo_id)
+
+    # check if the override config is different from the last override config,
+    # excluding the 'force_full' key.  otherwise using the --force-full flag
+    # would always trigger a full sync on the next sync too
+    prev_config = repo_importer.last_override_config.copy()
+    current_config = call_config.override_config.copy()
+
+    prev_config.pop('force_full', False)
+    current_config.pop('force_full', False)
```

Comment:
@ipanova I remove the key because of a corner case.  Consider using the "force_full" flag manually.  On the next sync, whether "force_full" is used or not, it will believe that the configuration has changed and force another full sync.